clifford.py

```diff
@@ -0,0 +1,58 @@
+from typing import Tuple, Any
+from abc import ABC, abstractmethod
+import uuid
+import numpy as np
+
+class AbstractProgramContext(ABC):
+    @abstractmethod
+    def add_gate_instruction(self, gate_name: str, target: Tuple[int], *params: Any):
+        pass
+
+class OpenQASMProgram:
+    def __init__(self, source: str, inputs: dict):
+        self.source = source
+        self.inputs = inputs
+
+class GateModelTaskResult:
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+class CliffordSimulator(OpenQASMSimulator):
```

Comment:
This doesn't seem to actually run a program; what the issue asks for is a concrete _implementation_ of a Clifford simulator

Reply:
At first glance, it looks like the paper you shared is for _circuit cutting_ rather than simulation. Good examples for the latter are [Stim](https://github.com/quantumlib/Stim) and Qiskit's [Clifford](https://docs.quantum.ibm.com/api/qiskit/qiskit.quantum_info.Clifford).